java/DoublyLinkedList.java

```diff
@@ -0,0 +1,123 @@
+public class DoublyLinkedList {
+
+  public Node head;
+  public Node tail;
+  public int length = 0;
+
+  private class Node {
+    private Node next;
+    private Node prev;
+    private String data;
+
+    // Constructor
+    public Node(Node initNext, Node initPrev, String initData){
+      this.next = initNext;
+      this.prev = initPrev;
+      this.data = initData;
+    }
+
+    // Getters
+    public Node getNext() {
+      return this.next;
+    }
+    public Node getPrev() {
+      return this.prev;
+    }
+    public String getData() {
+      return this.data;
+    }
+
+    // Setters
+    public void setNext(Node setTo) {
+      this.next = setTo;
+    }
+    public void setPrev(Node setTo) {
+      this.prev = setTo;
+    }
+    public void setData(String setTo) {
+      this.data = setTo;
+    }
+  } // End of Node class
+
+  public String insertNode(String data){
+    Node expo = new Node (null, null, data);
+    Node currentNode = this.head;
+
+    if(this.head == null){
+      this.head = expo;
+      this.tail = expo;
```

Comment:
Seems like lines 48-49 and 56-57 are doing the same things, regardless of what conditional branch they're in. Extract them out of the `if` `else`, since they're repeated in both.